src/components/menu/menu-item/menu-item.component.js

```diff
@@ -1,33 +1,211 @@
+  const actualFocusedItemIndex = useRef();
+  const submenuWrapperRef = useRef();
+  const setFocusToSubmenuElement = useCallback((event, index) => {
+    if (event) {
+      event.preventDefault();
+    }
+
+    const findIndex = () => {
+      if (submenuItemsRefs.current[index] && submenuItemsRefs.current[index].current.tagName === 'DIV') {
+        if (Events.isDownKey(event)) {
+          return index + 1;
+        }
+        return index - 1;
+      }
+      return index;
+    };
+
+    actualFocusedItemIndex.current = findIndex();
+
+    const isFocusOnLastMenuItem = index === submenuItemsRefs.current.length;
+    const isFocusOnFirstMenuItem = index === -1;
+    if (isFocusOnLastMenuItem) {
+      actualFocusedItemIndex.current = 0;
+    } else if (isFocusOnFirstMenuItem) {
+      actualFocusedItemIndex.current = submenuItemsRefs.current.length - 1;
+    }
+
+    submenuItemsRefs.current[actualFocusedItemIndex.current].current.focus();
+  }, [submenuItemsRefs]);
+
+  const onCloseSubmenu = useCallback(() => {
+    setIsOpenSubmenu(null);
+    setOpenByMouseEnter(false);
+    setOpenByArrowUp(false);
+  }, [setIsOpenSubmenu]);
+
+  const detectClickOutside = useCallback((e) => {
+    if (!Events.composedPath(e).includes(submenuWrapperRef.current)) {
+      onCloseSubmenu();
+      document.removeEventListener('click', detectClickOutside);
+    }
+  }, [onCloseSubmenu]);
+
+  useEffect(() => {
+    if (isOpen) {
+      document.addEventListener('click', detectClickOutside);
+    }
+    if (submenu && isOpen) {
+      if (!didOpenByArrowUp && !didOpenByMouseEnter && !didOpenByClick) {
+        submenuItemsRefs.current[0].current.focus();
+      } else if (didOpenByArrowUp && !didOpenByMouseEnter && !didOpenByClick) {
+        submenuItemsRefs.current[submenuItemsRefs.current.length - 1].current.focus();
+      }
+    }
+  // eslint-disable-next-line max-len
+  }, [didOpenByArrowUp, submenu, submenuItemsRefs, detectClickOutside, isOpen, didOpenByMouseEnter, onCloseSubmenu, setIsOpenSubmenu, menuItemIndex, ref, didOpenByClick]);
+
+  const onKeyDownSubmenu = useCallback((ev, index) => {
+    if (!isOpen) {
+      if (Events.isEnterKey(ev) || Events.isSpaceKey(ev) || Events.isDownKey(ev)) {
+        ev.preventDefault();
+        setIsOpenSubmenu(menuItemIndex);
+      }
+      if (Events.isUpKey(ev)) {
+        ev.preventDefault();
+        setOpenByArrowUp(true);
+        setIsOpenSubmenu(menuItemIndex);
+      }
+    }
+
+    if (isOpen) {
+      if (didOpenByClick) {
+        if (Events.isDownKey(ev)) {
+          ev.preventDefault();
+          setOpenByClick(false);
+          setFocusToSubmenuElement(ev, 0);
+        }
+
+        if (Events.isUpKey(ev)) {
+          setOpenByArrowUp(true);
+          ev.preventDefault();
+          setOpenByClick(false);
+          setFocusToSubmenuElement(ev, submenuItemsRefs.current.length - 1);
+        }
+      }
+      if (Events.isDownKey(ev) && !didOpenByClick) {
```

Comment:
Shouldnt this `if (Events.isDownKey(ev) && !didOpenByClick) {`
be put in 
`
if (didOpenByClick) {}
else { HERE } 
`